flink-runtime/src/main/java/org/apache/flink/runtime/state/ttl/TtlStateFactory.java

```diff
@@ -97,37 +96,35 @@ private TtlStateFactory(
         this.incrementalCleanup = getTtlIncrementalCleanup();
     }
 
-    private Map<Class<? extends StateDescriptor>, SupplierWithException<IS, Exception>>
-            createStateFactories() {
+    private Map<StateDescriptor.Type, SupplierWithException<IS, Exception>> createStateFactories() {
         return Stream.of(
                         Tuple2.of(
-                                ValueStateDescriptor.class,
+                                StateDescriptor.Type.VALUE,
                                 (SupplierWithException<IS, Exception>) this::createValueState),
```

Comment:
IIUC, this mathe is followed by a similar match in passed keyed state backend (e.g. `HeapKeyedStateBackend`).
(in `TtlStateFactory.createTtlStateContext`)
Currently, those backends use similar code:
- `HeapKeyedStateBackend.STATE_FACTORIES`
- `RocksDBKeyedStateBackend.STATE_FACTORIES`
- `ChangelogKeyedStateBackend.STATE_FACTORIES`

Shouldn't they also be updated?

Reply:
Thanks for the kind reminder, I'll update all related factories and rename the original ticket and this PR.